### PR TITLE
Return 204 from the Sync Route

### PR DESCRIPTION
**Pull request by Magnus-Kuhn**
# Readiness checklist

-   [x] I added/updated tests.
-   [x] I ensured that the PR title is good enough for the changelog.
-   [x] I labeled the PR.

### CODE REVIEW

File: src/modules/coreHttpApi/controllers/AccountController.ts [hunk 1]
```diff
@@ -23,7 +24,8 @@ export class AccountController extends BaseController {
     @Accept("application/json")
     public async sync(): Promise<Envelope> {
         const result = await this.transportServices.account.syncEverything();
-        return this.ok(result);
+        if (result.isError) return this.ok(result);
```
Comment (jkoenig134) [inline src/modules/coreHttpApi/controllers/AccountController.ts:27]:
```suggestion
        if (result.isError) throw result.error;
```
this is what would happen deep in the this.ok call :)

File: src/modules/coreHttpApi/controllers/AccountController.ts [hunk 2]
```diff
@@ -23,7 +24,8 @@ export class AccountController extends BaseController {
...
+        return this.ok(Result.ok("Sync successful"));
```
Comment (jkoenig134) [inline src/modules/coreHttpApi/controllers/AccountController.ts:28]: I don't know if this was discussed or not, but IMO `"Sync successful"` is not a good return value.. wtb just making it a 204?

Comment (Magnus-Kuhn) [reply src/modules/coreHttpApi/controllers/AccountController.ts:28]: I just followed the Confluence page, I agree with 204

File: packages/sdk/src/endpoints/AccountEndpoint.ts [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
 import { Endpoint } from "./Endpoint";
 
 export class AccountEndpoint extends Endpoint {
     public async getIdentityInfo(): Promise<ConnectorResponse<IdentityInfo>> {
         return await this.get("/api/v2/Account/IdentityInfo");
     }
 
-    public async sync(): Promise<ConnectorResponse<ConnectorSyncResult>> {
-        return await this.post("/api/v2/Account/Sync", undefined, 200);
+    public async sync(): Promise<ConnectorResponse<string>> {
```
Comment (jkoenig134) [inline packages/sdk/src/endpoints/AccountEndpoint.ts:9]: void

File: src/modules/coreHttpApi/openapi.yml [hunk 4]
```diff
@@ -238,33 +238,11 @@ paths:
...
-                        type: array
-                        items:
-                          $ref: "#/components/schemas/Message"
-                    required:
-                      - messages
-                      - relationships
-                required:
-                  - result
+        204:
+          description: Success. Received new/changed Relationships/Messages, if present.
```
Comment (jkoenig134) [inline src/modules/coreHttpApi/openapi.yml:245]:
```suggestion
          description: No content.
```

Comment (Magnus-Kuhn) [reply src/modules/coreHttpApi/openapi.yml:245]: I prefer the style of your recent 204 description of the attribute deletion which actually explains something.

Comment (jkoenig134) [reply src/modules/coreHttpApi/openapi.yml:245]: I don't.

Comment (jkoenig134) [reply src/modules/coreHttpApi/openapi.yml:245]: Especially if your description is only partially right.

Comment (jkoenig134) [reply src/modules/coreHttpApi/openapi.yml:245]: As is the description is very misleading.

Comment (Magnus-Kuhn) [reply src/modules/coreHttpApi/openapi.yml:245]: How?

Comment (jkoenig134) [reply src/modules/coreHttpApi/openapi.yml:245]: Sounds like you would receive new/changed Relationships/Messages, but you don't receive anything.

File: src/modules/coreHttpApi/openapi.yml [hunk 5]
```diff
@@ -238,33 +238,11 @@ paths:
...
-                        type: array
-                        items:
-                          $ref: "#/components/schemas/Message"
-                    required:
-                      - messages
-                      - relationships
-                required:
-                  - result
+        204:
+          description: No content. The Connector received new/changed Relationships/Messages from the Backbone, if present.
```
Comment (jkoenig134) [inline src/modules/coreHttpApi/openapi.yml:245]: still wrong. Please just remove the extra text.....

**Review by jkoenig134 — APPROVED** [review]

**Pull request merged**